docs/HowToFAQ/Errors-Resolution.md

```diff
@@ -423,34 +431,77 @@ are ORDINALs that are larger than any `nat`. Logical reasoning with ORDINALs is
 a bit counter-intuitive at times. For logical implementation reasons, Dafny limits where
 ORDINALs can be used; one restriction is that the ORDINAL type may not be a type argument.
 
-<!-- ./DafnyCore/Resolver/Abstemious.cs-->
+<!-- FILE ./DafnyCore/Resolver/Abstemious.cs -->
+
+## **Error: the value returned by an abstemious function must come from invoking a co-constructor** {#r_abstemious_needs_conconstructor}
 
-## **the value returned by an abstemious function must come from invoking a co-constructor**
+```dafny
+codatatype D = A | B
+function {:abstemious} f(): int {0}
+```
 
 <!-- TODO -->
+_Abstemious functions are not documented. Please report occurences of this error message._
```

Comment:
That's a relief, first I've ever heard of them :)

Reply:
But they are a thing (related to codatatypes) -- Rustan has a task to add a bit of documentation.